src/apps/shared/query_params.py

```diff
@@ -0,0 +1,54 @@
+from typing import Any
+
+from fastapi import Depends
+from pydantic import BaseModel, Field
+
+
+class BaseQueryParams(BaseModel):
+    """
+    Class to declare query parameters
+    """
+
+    search: str | None = None
+    page: int = Field(gt=0, default=10)
+    limit: int = Field(gt=0, default=10)
+    ordering: str | None = None
+
+
+class QueryParams(BaseModel):
+    """
+    Class to store grouped query parameters
+    """
+
+    filters: dict[str, Any] = Field(default_factory=dict)
+    search: str | None = None
+    page: int = Field(gt=0, default=10)
+    limit: int = Field(gt=0, default=10)
+    ordering: list[str] = Field(default_factory=list)
```

Comment:
Why this one is shared for every module?
Are we going to use the same query parameters for any of our submodules?

Even if yes I think we should go out from such general naming like `QueryParams`. What about `PaginationQueryParams` or something like that?

Also for some reason, you have the `BaseQueryParams` that I am expecting you should inherit from but it is not. Could we maybe rename it as far as it is not a really `Base...` class